libursa/src/encryption/symm/aescbc.rs

```diff
@@ -0,0 +1,141 @@
+use super::Encryptor;
+use aead::{
+    generic_array::{
+        typenum::{Unsigned, U0, U16, U32, U48, U64},
+        GenericArray,
+    },
+    Aead, Error, NewAead, Payload,
+};
+use aes::{Aes128, Aes256};
+use block_modes::block_padding::Pkcs7;
+use block_modes::{BlockMode, Cbc};
+use hmac::{Hmac, Mac};
+#[cfg(feature = "serialization")]
+use serde::de::{Deserialize, Deserializer, Error as DError, Visitor};
+#[cfg(feature = "serialization")]
+use serde::ser::{Serialize, Serializer};
+use sha2::{Sha256, Sha512};
+use subtle::ConstantTimeEq;
+use zeroize::Zeroize;
+
+type Aes128Cbc = Cbc<Aes128, Pkcs7>;
+type Aes256Cbc = Cbc<Aes256, Pkcs7>;
+type HmacSha256 = Hmac<Sha256>;
+type HmacSha512 = Hmac<Sha512>;
+
+macro_rules! aes_cbc_hmac_impl {
+    ($name:ident, $algokeysize:ident, $keysize:ident, $noncesize:ident, $tagsize:ident, $algo:ident, $hash:ident, $visitor:ident) => {
+        #[derive(Debug, Clone, Eq, PartialEq)]
+        pub struct $name {
+            key: GenericArray<u8, $keysize>,
+        }
+
+        impl Encryptor for $name {
+            type MinSize = U48;
+        }
+
+        impl NewAead for $name {
+            type KeySize = $keysize;
+
+            fn new(key: GenericArray<u8, $keysize>) -> Self {
+                Self { key }
+            }
+        }
+
+        impl Aead for $name {
+            type NonceSize = $noncesize;
+            type TagSize = $tagsize;
+            type CiphertextOverhead = U0;
+
+            fn encrypt<'msg, 'aad>(
```

Comment:
Some documentation of `encrypt` and `decrypt` would be helpful?

Reply:
Like what. Is it not clear how to use encrypt?

Reply:
The logic of `encrypt` and `decrypt`

Reply:
I don't think its necessary. Anyone crypto minded knows what encrypt and decrypt do.